### PR TITLE
Metrics: measurement mapping by regex

### PR DESCRIPTION
**Pull request by sirtik**

### CODE REVIEW

**Comment by sirtik** [comment]
ping @nzroller

File: metrics-influxdb/src/main/java/com/izettle/metrics/influxdb/InfluxDbReporter.java [hunk 1]
```diff
@@ -24,6 +25,7 @@
         private boolean groupGauges;
         private Set<String> includeTimerFields;
         private Set<String> includeMeterFields;
+        private Map<String, String> measurementMappings;
```
Comment (nzroller) [inline metrics-influxdb/src/main/java/com/izettle/metrics/influxdb/InfluxDbReporter.java]: Let's make this a `Map<String, Pattern>`

Comment (sirtik) [inline metrics-influxdb/src/main/java/com/izettle/metrics/influxdb/InfluxDbReporter.java]: Alright

File: metrics-influxdb/src/test/java/com/izettle/metrics/influxdb/InfluxDbReporterTest.java [hunk 2]
```diff
@@ -235,6 +235,56 @@ public void reportsIncludedMeters() throws Exception {
...
+            this.<Histogram>map(),
+            this.map("com.example.resources.RandomResource", mock(Meter.class)),
+            this.<Timer>map()
+        );
+
+        final ArgumentCaptor<InfluxDbPoint> influxDbPointCaptor = ArgumentCaptor.forClass(InfluxDbPoint.class);
+        Mockito.verify(influxDb, atLeastOnce()).appendPoints(influxDbPointCaptor.capture());
+        InfluxDbPoint point = influxDbPointCaptor.getValue();
+
+        assertThat(point.getMeasurement()).isEqualTo("resources");
```
Comment (nzroller) [inline metrics-influxdb/src/test/java/com/izettle/metrics/influxdb/InfluxDbReporterTest.java]: cool, check that it's tagged too! e.g. metricName=com.example.resources.RandomResource

File: metrics-influxdb/src/main/java/com/izettle/metrics/influxdb/InfluxDbReporter.java [hunk 3]
```diff
@@ -136,10 +138,22 @@ public Builder includeMeterFields(Set<String> fields) {
...
         }
 
+        /**
+         * Map measurement to a defined measurement name, where the key is the measurement name
+         * and the value is the reqex the measurement should be mapped by.
+         *
+         * @param measurementMappings
+         * @return {@code this}
+         */
+        public Builder measurementMappings(Map<String, String> measurementMappings) {
```
Comment (nzroller) [inline metrics-influxdb/src/main/java/com/izettle/metrics/influxdb/InfluxDbReporter.java]: To make it easy to use, probably construct the Map<String, Pattern> here and throw a runtime if a pattern doesn't compile?

Comment (sirtik) [inline metrics-influxdb/src/main/java/com/izettle/metrics/influxdb/InfluxDbReporter.java]: I like the idea of sending in the map with pattern values. Then we don't need to construct another map here and the caller should be sure, that they are sending in patterns that compile too.

**Comment by sirtik** [comment]
@nzroller changed the measurementMappings to <String,Pattern> and added metricName tag to points.

File: metrics-influxdb/src/main/java/com/izettle/metrics/influxdb/InfluxDbReporter.java [hunk 4]
```diff
@@ -136,10 +138,22 @@ public Builder includeMeterFields(Set<String> fields) {
...
         }
 
+        /**
+         * Map measurement to a defined measurement name, where the key is the measurement name
+         * and the value is the reqex the measurement should be mapped by.
+         *
+         * @param measurementMappings
+         * @return {@code this}
+         */
+        public Builder measurementMappings(Map<String, Pattern> measurementMappings) {
```
Comment (nzroller) [inline metrics-influxdb/src/main/java/com/izettle/metrics/influxdb/InfluxDbReporter.java]: Cool, though I think from a usability stand-point we should pass in `Map<String, String>` and compile the patterns in this method.

File: metrics-influxdb/src/main/java/com/izettle/metrics/influxdb/InfluxDbReporter.java [hunk 5]
```diff
@@ -430,4 +466,20 @@ private long calculateDelta(String name, long count) {
         return count - previous;
     }
 
+    private String getMeasurementName(final String name) {
+
+        if (measurementMappings == null) {
```
Comment (nzroller) [inline metrics-influxdb/src/main/java/com/izettle/metrics/influxdb/InfluxDbReporter.java]: Probably just initialize to an empty map?

**Comment by sirtik** [comment]
@nzroller

**Comment by sirtik** [comment]
and again :D @nzroller

**Comment by nzroller** [comment]
Thanks @sirtik !

**Pull request merged**